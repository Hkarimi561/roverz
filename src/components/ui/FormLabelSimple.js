/**
 * Text Input
 *
     <FormLabel></FormLabel>
 *
 * React Native Starter App
 * https://github.com/mcnamee/react-native-starter-app
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormLabel } from 'react-native-elements';

// Consts and Libs
import { AppColors, AppFonts } from '../../theme/';

/* Component ==================================================================== */
class FormLabelSimple extends Component {
  static propTypes = {
    labelStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
    children: PropTypes.node,
  }

  static defaultProps = {
    containerStyle: [],
    labelStyle: [],
    children: null,
  }

  labelProps = () => {
    // Defaults
    let props = {
      ...this.props,
      labelStyle: [{
        color: AppColors.brand().secondary,
        fontFamily: AppFonts.base.family,
      }],
    };
    props = [props, this.props.labelStyle];
    return props;
  }

  render = () => <FormLabel {...this.labelProps()}>{this.props.children}</FormLabel>;
}

/* Export Component ==================================================================== */
export default FormLabelSimple;
