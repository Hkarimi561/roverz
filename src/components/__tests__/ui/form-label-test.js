/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import FormLabel from '@ui/FormLabel';
import FormLabelSimple from '@ui/FormLabelSimple';

it('FormLabel renders correctly', () => {
  const tree = renderer.create(
    <FormLabel>John Smith</FormLabel>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('FormLabel renders correctly', () => {
  const tree = renderer.create(
    <FormLabelSimple>John Smith</FormLabelSimple>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
