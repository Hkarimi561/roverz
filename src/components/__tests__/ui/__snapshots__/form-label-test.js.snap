// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel renders correctly 1`] = `
<View
  0={
    Object {
      "children": "John Smith",
      "containerStyle": Array [],
      "labelStyle": Array [
        Object {
          "color": "#222222",
          "fontFamily": "OpenSans-Regular",
          "marginLeft": 0,
          "marginRight": 0,
        },
      ],
    }
  }
  1={Array []}
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 15,
          },
          undefined,
          undefined,
        ],
      ]
    }
  >
    John Smith
  </Text>
</View>
`;

exports[`FormLabel renders correctly 2`] = `
<View
  0={
    Object {
      "children": "John Smith",
      "containerStyle": Array [],
      "labelStyle": Array [
        Object {
          "color": "#50527F",
          "fontFamily": "OpenSans-Regular",
        },
      ],
    }
  }
  1={Array []}
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 15,
          },
          undefined,
          undefined,
        ],
      ]
    }
  >
    John Smith
  </Text>
</View>
`;
